### ISSUE TITLE
compareScenarios does not accept years > 2005

### ISSUE THREAD

**Issue by Loisel** [pik-piam/remind2#8]
Here is my test case:
```
download.file("https://rse.pik-potsdam.de/data/example/fulldata_REMIND21.gdx",
              "fulldata.gdx", mode="wb")
remind2::convGDX2MIF("fulldata.gdx", file="out.mif")
remind2::compareScenarios(
  "out.mif",
  "~/git/REMIND/core/input/historical/historical.mif", y=seq(2010, 2055, 5), y_bar = c(2015, 2030, 2050),
  fileName="compareScenarioDefault_shortTerm.pdf")
```
it turns out that:
```
> remind2::compareScenarios(
+   "out.mif",
+   "~/git/REMIND/core/input/historical/historical.mif", y=seq(2010, 2055, 5), y_bar = c(2015, 2030, 2050),
+   fileName="compareScenarioDefault_shortTerm.pdf")
Error in x@.Data[i, j, k, drop = FALSE] : subscript out of bounds
> traceback()
6: .local(x, i, j, ..., drop)
5: kaya[, ref_year, ]
4: kaya[, ref_year, ]
3: setYears(kaya[, ref_year, ], NULL) at calcKayaDecomp.R#62
2: calcKayaDecomp(mif = data[, , i]) at compareScenarios.R#1347
1: remind2::compareScenarios("out.mif", "~/git/REMIND/core/input/historical/historical.mif", 
       y = seq(2010, 2055, 5), y_bar = c(2015, 2030, 2050), fileName = "compareScenarioDefault_shortTerm.pdf")
```
adding 2005 to the year range fixes this issue. Either this should be documented and corresponding check introduced for the parameter or fixed to work with ranges excluding 2005.

**Comment by Renato-Rodrigues** [comment]
From the looks of the bug report, the problem comes from the calcKayaDecomp function call, which requires a reference year that when not assigned is set to 2005 automatically.

You could try to set it to another year if 2005 is absent, but I am not sure about the importance of removing the 2005 year from the scenario comparison pdf, as it is our starting year for REMIND and it is always present anyway. I also never dealt with the Kaya decomposition report so maybe somebody else should give a more informed opinion about the preferable base year for it.

Alternatively, you could just add a warning in the beginning of the function, saying that 2005 is a required year, and add it automatically to the y parameter if not present in the user defined call of the compareScenarios function.

**Comment by chroetz** [comment]
In `compareScenarios2()` 2005 is still the hard-coded reference year for kaya decomposition. I will add a fall back (or additional YAML parameter) in case this is missing

**Comment by chroetz** [comment]
This issue is solved with #221. Note that the default value for the reference year for relative Kaya decomposition is now 2020.
